### PR TITLE
[go migration] More things that the migration tools and skill could convert

### PR DESCRIPTION
**Pull request by aiuto**
[go migration] More things that the migration tools and skill could convert

**what this does**
Migrates:
- comp/collector/collector/def
- comp/core/sysprobeconfig/impl
- comp/networkpath/traceroute/impl-remote
- comp/snmpscan
- comp/snmpscan/mock
- partial pkg/discovery/tracermetadata
- pkg/security/ebpf/kernel
- pkg/snmp
- pkg/snmp/devicededuper
- pkg/snmp/snmpparse
- pkg/system-probe/config

**Notes**

After these, the tools stop finding anything that works. Everything needs //pkg/ebpf.

### CODE REVIEW

File: pkg/fleet/installer/packages/embedded/BUILD.bazel [hunk 1]
```diff
@@ -106,6 +106,7 @@ go_test(
         "tmpl/gen/debrpm-nocap/**",
         "tmpl/gen/oci-nocap/**",
     ]),
+    tags = ["manual"],
```
[the remark this answers is not in the document]
Comment (aiuto) [reply pkg/fleet/installer/packages/embedded/BUILD.bazel]: This is the recommended solution from the rules_go maintainers.
https://github.com/bazel-contrib/rules_go/issues/3203

File: pkg/discovery/tracermetadata/BUILD.bazel [hunk 2]
```diff
@@ -1 +1,34 @@
...
+            "//pkg/discovery/tracermetadata/model",
+            "//pkg/util/kernel",
+        ],
+        "//conditions:default": [],
+    }),
+)
+
+go_test(
+    name = "tracermetadata_test",
+    srcs = ["tracer_java_test.go"],
```
[the remark this answers is not in the document]
Comment (aiuto) [reply pkg/discovery/tracermetadata/BUILD.bazel:25]: That's OK for now. We are building an separate process to validate that we have covered all the existing tests with bazel. That will catch these edge cases so we can address them later.  The priority today is to make the go_library targets available for other packages to migrate.

**Review by chouquette — APPROVED** [review]
